core/templates/local_vector.h

```diff
@@ -334,6 +344,26 @@ class LocalVector {
 			data[i] = p_from[i];
 		}
 	}
+	inline void operator=(LocalVector &&p_from) {
+		if (this == &p_from) {
+			return;
+		}
+		reset();
+
+		data = p_from.data;
+		count = p_from.count;
+		capacity = p_from.capacity;
+
+		p_from.data = nullptr;
+		p_from.count = 0;
+		p_from.capacity = 0;
+	}
+	inline void operator=(Vector<T> &&p_from) {
+		resize(p_from.size());
+		for (U i = 0; i < count; i++) {
+			data[i] = std::move(p_from[i]);
```

Comment:
I think this leave the vector in an invalid state, that is it still has a size, but all of its elements are moved out. The vector needs to be left in an "undefined but valid" state.

Reply:
I think `p_from.clear()` would do the trick; that just resizes to 0, which has a dedicated cleanup routine in CowData:
https://github.com/godotengine/godot/blob/fafc07335bdecacd96b548c4119fbe1f47ee5866/core/templates/cowdata.h#L337-L342

Reply:
I don't think tue Code ist wrong.
std::move guarantees that the argument is left in a valid state. All elements of the argument vector have been moved out of the vector, leaving them in a valid state. That means the vector itself is left in a valid state.
The array does not have to be destructed, because std::move does not guarantee the argument is explicitly destructed.

Reply:
Right, yeah, the Vector<T> will have a bunch of T's in it which are themselves in a valid but undefined state. So it is still a valid Vector<T> of size n. Okay, you convinced me.